### PR TITLE
TE Cancellation

### PR DESCRIPTION
**Pull request by DadImScared**

### CODE REVIEW

File: frontend/src/components/TE/TaskMixin.ts [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+    item: any,
+    itemPath: string,
+    paths: any,
+    fieldUpdates: any
+  ) {
+    try {
+      await firebase.functions().httpsCallable("TE-cancelAllotment")({
+        taskId: item[".key"]
+      });
+      this.updateFields(item, fieldUpdates);
```
Comment (anantakrishna) [inline frontend/src/components/TE/TaskMixin.ts]: We don't need to update the database from the frontend. It is supposed that all this is done by the function. So you need to call the function only.

File: frontend/src/views/TE/Task.vue [hunk 2]
```diff
@@ -135,6 +137,14 @@ export default class Task extends Mixins<TaskMixin, FormatTime>(
     this.isCoordinator = claims.coordinator;
   }
 
+  onCancelClick() {
+    this.cancelAllotment(this.task, "", {}, this.cancelData());
```
Comment (anantakrishna) [inline frontend/src/views/TE/Task.vue]: This call looks ugly. Isn't it possible to have this method `cancelAllotment` without any parameters and be directly hooked to button's `@click`? As all the necessary info for cancellation is available in the context. In other words, we are not calling this method with other set of parameters, and are not going to.

Comment (DadImScared) [reply frontend/src/views/TE/Task.vue]: Currently the cancel callback is shared between tasks view and task view. Tasks view uses the assignee callback which gets called with those parameters. Should the cancel callback be unique for the separate views?

File: frontend/src/components/TE/TaskMixin.ts (hunk 1 above)
Comment (DadImScared) [reply frontend/src/components/TE/TaskMixin.ts]: That function only updates the local state on a successful https function call.

Comment (anantakrishna) [reply frontend/src/components/TE/TaskMixin.ts]: Ok then

File: frontend/src/views/TE/Task.vue (hunk 2 above)
Comment (anantakrishna) [reply frontend/src/views/TE/Task.vue]: I think in this case we can have them separate

**Pull request merged**